### PR TITLE
Filter components tests

### PR DESCRIPTION
**Pull request by jtkpiotr**
@adamzima @quirk0o @pbzymek @s1mplex @lkudyba @mhib

### CODE REVIEW

File: client/test/FilterComponent-test.jsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import {Checkbox} from 'react-bootstrap'
+import {mount, shallow} from 'enzyme';
+import Filter from '../app/components/calendar/filters/Filter'
+import chai from 'chai'
+import jsdom from 'mocha-jsdom'
+import sinon from 'sinon'
+
+var expect = chai.expect;
```
Comment (quirk0o) [inline client/test/FilterComponent-test.jsx]: You can do

```
import { expect } from 'chai';
```

File: client/test/FilterComponent-test.jsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    conferenceRoom,
+    onEnabled: () => {
+    },
+    onDisabled: () => {
+    },
+    enabled: false
+  };
+  const defaultWrapper = shallow(<Filter {...defaultProps} />);
+
+  it('renders Checkbox component', () => {
```
Comment (quirk0o) [inline client/test/FilterComponent-test.jsx]: Convention is to write `renders <Checkbox />`

File: client/test/filters/RoomFiltersComponent.spec.jsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  it('renders appropriate number of <Filter />', () => {
+    expect(defaultWrapper.find(Filter)).to.have.length(conferenceRooms.length);
+  });
+
+  describe('_filterEnabled', () => {
+    it('checks if conferenceRoom is filtered', () => {
+      defaultWrapper.setProps({ filters: [id1, id2] });
+      expect(defaultWrapper.instance()._filterEnabled(conferenceRoom1)).to.eq(true);
+      expect(defaultWrapper.instance()._filterEnabled(conferenceRoom2)).to.eq(true);
+      expect(defaultWrapper.instance()._filterEnabled(conferenceRoom3)).to.eq(false);
```
Comment (quirk0o) [inline client/test/filters/RoomFiltersComponent.spec.jsx]: The naming convention with `_` implies that this is a method private to the component. That means that it is likely to change and I don't think you should be basing your tests on such a method.

File: client/test/filters/RoomFiltersComponent.spec.jsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  const props = {
+    conferenceRooms,
+    onEnabled:  () => {},
+    onDisabled: () => {}
+  };
+  const defaultWrapper = shallow(<RoomFilters {...props} />);
+
+  describe('#render()', () => {
+    it('renders .filter-container', () => {
+      expect(defaultWrapper.find('.filter-container')).to.have.length(1);
```
Comment (mhib) [inline client/test/filters/RoomFiltersComponent.spec.jsx]: lengthOf

File: client/test/filters/FilterComponent.spec.jsx [hunk 5]
```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import { Checkbox } from 'react-bootstrap';
+import { mount, shallow } from 'enzyme';
+import Filter from '../../app/components/calendar/filters/Filter';
+import chai from 'chai';
+import sinon from 'sinon';
+import ConferenceRoom from '../factories/ConferenceRoom';
+
+const expect = chai.expect;
```
Comment (quirk0o) [inline client/test/filters/FilterComponent.spec.jsx]: Use `import { expect } from 'chai';` instead :)

File: client/test/filters/FilterComponent.spec.jsx [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+describe('<Filter />', () => {
+  const color = '#000000';
+  const title = 'sample_title';
+  const id = 1;
+  const capacity = 1;
+  const conferenceRoom = ConferenceRoom.build({ id, title, color, capacity });
```
Comment (quirk0o) [inline client/test/filters/FilterComponent.spec.jsx]: You can use the default values of the factory, no need to override attributes :)

File: client/test/filters/FilterComponent.spec.jsx [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+  it('sets .filter-box backgroundColor', () => {
+    expect(defaultWrapper.find('.filter-box').props().style.backgroundColor).to.eq(color);
+  });
+
+  it('puts conference room title inside Checkbox component', () => {
+    expect(defaultWrapper.find(Checkbox).children().text()).to.eq(title);
+  });
+
+  it("triggers appropriate handler based on 'enabled' prop", () => {
```
Comment (quirk0o) [inline client/test/filters/FilterComponent.spec.jsx]: Double quotes

File: client/test/filters/FilterComponent.spec.jsx [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+    const props = {
+      conferenceRoom,
+      onEnabled,
+      onDisabled,
+      enabled: false
+    };
+    const wrapper = mount(<Filter {...props} />);
+
+    wrapper.find('input').simulate('change');
+    expect(onDisabled).to.have.property('callCount', 1);
```
Comment (quirk0o) [inline client/test/filters/FilterComponent.spec.jsx]: You can use `sinon-chai` library assertions and write `expect(onDisabled).to.have.been.calledOnce()`

File: client/test/filters/FilterComponent.spec.jsx [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+    expect(onDisabled.calledWith(id)).to.eq(true);
```
Comment (quirk0o) [inline client/test/filters/FilterComponent.spec.jsx]: Also `sinon-chai` provides `expect(onDisabled).to.have.been.calledWith(id)`

File: client/test/filters/RoomFiltersComponent.spec.jsx [hunk 10]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import Filter from '../../app/components/calendar/filters/Filter';
+import RoomFilters from '../../app/components/calendar/filters/RoomFilters';
+import chai from 'chai';
+import ConferenceRoom from '../factories/ConferenceRoom';
+
+const expect = chai.expect;
```
Comment (quirk0o) [inline client/test/filters/RoomFiltersComponent.spec.jsx]: Same here

File: client/test/filters/RoomFiltersComponent.spec.jsx [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+
+describe('<RoomFilters />', () => {
+  const [id1, id2, id3] = [1, 2, 3];
```
Comment (quirk0o) [inline client/test/filters/RoomFiltersComponent.spec.jsx]: ConferenceRoom factory automatically increments ids so no need to do this

File: client/test/filters/RoomFiltersComponent.spec.jsx [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+  const [title1, title2, title3] = ['title1', 'title2', 'title3'];
+  const conferenceRoom1 = ConferenceRoom.build({ id: id1, title: title1 });
+  const conferenceRoom2 = ConferenceRoom.build({ id: id2, title: title2 });
+  const conferenceRoom3 = ConferenceRoom.build({ id: id3, title: title3 });
```
Comment (quirk0o) [inline client/test/filters/RoomFiltersComponent.spec.jsx]: Do you actually need the conference rooms to have different titles?

File: client/test/filters/RoomFiltersComponent.spec.jsx [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+import Filter from '../../app/components/calendar/filters/Filter';
+import RoomFilters from '../../app/components/calendar/filters/RoomFilters';
+import chai from 'chai';
+import ConferenceRoom from '../factories/ConferenceRoom';
+
+const expect = chai.expect;
+
+describe('<RoomFilters />', () => {
+  const [id1, id2, id3] = [1, 2, 3];
+  const [title1, title2, title3] = ['title1', 'title2', 'title3'];
```
Comment (quirk0o) [inline client/test/filters/RoomFiltersComponent.spec.jsx]: Do you actually need the conference rooms to have different titles?

File: client/test/filters/FilterComponent.spec.jsx [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+
+  it('sets .filter-box backgroundColor', () => {
+    expect(defaultWrapper.find('.filter-box').props().style.backgroundColor).to.eq(conferenceRoom.color);
+  });
+
+  it('puts conference room title inside Checkbox component', () => {
+    expect(defaultWrapper.find(Checkbox).children().text()).to.eq(conferenceRoom.title);
+  });
+
+  it("triggers appropriate handler based on 'enabled' prop", () => {
```
Comment (madejejej) [inline client/test/filters/FilterComponent.spec.jsx]: I'd go with 2 test cases, one for enabled and one for disabled

File: client/test/filters/RoomFiltersComponent.spec.jsx [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+import { shallow } from 'enzyme';
+import Filter from '../../app/components/calendar/filters/Filter';
+import RoomFilters from '../../app/components/calendar/filters/RoomFilters';
+import { expect } from 'chai';
+import ConferenceRoom from '../factories/ConferenceRoom';
+
+describe('<RoomFilters />', () => {
+  const conferenceRoom1 = ConferenceRoom.build();
+  const conferenceRoom2 = ConferenceRoom.build();
+  const conferenceRoom3 = ConferenceRoom.build();
```
Comment (madejejej) [inline client/test/filters/RoomFiltersComponent.spec.jsx]: You could use `ConferenceRoom.buildList(3)` instead

**Pull request merged**